hushline/premium.py

```diff
@@ -0,0 +1,536 @@
+            .all()
+        )
+
+        return render_template("premium.html", user=user, invoices=invoices)
+
+    @bp.route("/waiting", methods=["GET"])
+    @authentication_required
+    def waiting() -> Response | str:
+        return render_template("premium-waiting.html")
+
+    @bp.route("/upgrade", methods=["GET", "POST"])
+    @authentication_required
+    def upgrade() -> Response | str:
+        if request.method == "GET":
+            return redirect(url_for("premium.index"))
+
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        # If the user is already on the business tier
+        if user.tier_id == BUSINESS_TIER:
+            flash("👍 You're already upgraded.")
+            return redirect(url_for("premium.index"))
+
+        # Select the business tier
+        business_tier = db.session.query(Tier).get(BUSINESS_TIER)
+        if not business_tier:
+            current_app.logger.error("Could not find business tier")
+            flash("⚠️ Something went wrong!")
+            return redirect(url_for("premium.index"))
+
+        # Does the user already have a Stripe customer?
+        if not user.stripe_customer_id:
+            try:
+                create_customer(user)
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Failed to create Stripe customer: {e}")
+                flash("⚠️ Something went wrong!")
+                return redirect(url_for("premium.index"))
+
+        # Create a Stripe Checkout session
+        try:
+            checkout_session = stripe.checkout.Session.create(
+                client_reference_id=str(user.id),
+                customer=user.stripe_customer_id,
+                line_items=[{"price": business_tier.stripe_price_id, "quantity": 1}],
+                mode="subscription",
+                success_url=url_for("premium.waiting", _external=True),
+                automatic_tax={"enabled": True},
+                customer_update={"address": "auto"},
+            )
+        except stripe._error.StripeError as e:
+            current_app.logger.error(f"Failed to create Stripe Checkout session: {e}")
+            return abort(500)
+
+        if checkout_session.url:
+            return redirect(checkout_session.url)
+
+        return abort(500)
+
+    @bp.route("/disable-autorenew", methods=["POST"])
+    @authentication_required
+    def disable_autorenew() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                stripe.Subscription.modify(user.stripe_subscription_id, cancel_at_period_end=True)
+                user.stripe_subscription_cancel_at_period_end = True
```

Comment:
the `try/except` should only be around the `modify` call, the rest should come after or be in an `else` block

Reply:
Fixed in ff5f2173ca30559812e5b4a9652b6355b061200a